frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -24,69 +34,119 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
+      getPluginPatch(consoleOperatorConfig, plugin, false),
+    );
+
+    const allPromises = [
+      k8sKill(SubscriptionModel, subscription, {}, deleteOptions),
+      ...(subscription?.status?.installedCSV
+        ? [
+            k8sKill(
+              ClusterServiceVersionModel,
+              {
+                metadata: {
+                  name: subscription.status.installedCSV,
+                  namespace: subscription.metadata.namespace,
+                },
+              },
+              {},
+              deleteOptions,
+            ).catch(() => Promise.resolve()),
+          ]
+        : []),
+    ];
+    canPatchConsoleOperatorConfig &&
+      allPromises.push(k8sPatch(ConsoleOperatorConfigModel, consoleOperatorConfig, patch));
+
+    handlePromise(Promise.all(allPromises))
+      .then(() => {
+        close();
+        if (
+          window.location.pathname.split('/').includes(subscription.metadata.name) ||
+          window.location.pathname.split('/').includes(subscription.status.installedCSV)
+        ) {
+          history.push(resourceListPathFromModel(ClusterServiceVersionModel, getActiveNamespace()));
+        }
+      })
+      .catch(() => {});
+  };
+
   const name = csv?.spec?.displayName || subscription?.spec?.name;
   const namespace =
     subscription.metadata.namespace === GLOBAL_OPERATOR_NAMESPACE
       ? 'all-namespaces'
       : subscription.metadata.namespace;
   const uninstallMessage = csv?.metadata?.annotations?.[OPERATOR_UNINSTALL_MESSAGE_ANNOTATION];
+  const pluginName = getPlugins();
   return (
     <form onSubmit={submit} name="form" className="modal-content co-catalog-install-modal">
       <ModalTitle className="modal-header">
         <YellowExclamationTriangleIcon className="co-icon-space-r" /> {t('olm~Uninstall Operator?')}
       </ModalTitle>
       <ModalBody>
-        <Trans t={t} ns="olm">
-          This will remove Operator <strong>{{ name }}</strong> from{' '}
-          <strong>{{ namespace }}</strong>. Removing the Operator will not remove any of its custom
-          resource definitions or managed resources. If your Operator has deployed applications on
-          the cluster or configured off-cluster resources, these will continue to run and need to be
-          cleaned up manually.
-        </Trans>
+        <p>
+          <Trans t={t} ns="olm">
+            Operator <strong>{{ name }}</strong> will be removed from{' '}
+            <strong>{{ namespace }}</strong>, but any of its custom resource definitions or managed
+            resources will remain. If your Operator deployed applications on the cluster or
+            configured off-cluster resources, these will continue to run and require manual cleanup.
+          </Trans>
+        </p>
+        {canPatchConsoleOperatorConfig && enabledPlugins.length === 1 && (
+          <p>
+            <Trans t={t} ns="olm">
+              The <strong>{{ pluginName }}</strong> console plugin provided by this operator will be
```

Comment:
This is going to bold the `and` in the plugin label.

Reply:
We might just skip showing the plugin name. It's generally not too important and could cause problems for translation.

Reply:
More than happy to drop that.